pynio/service.py

```diff
@@ -70,11 +72,31 @@ def clean(connection):
 
     def start(self):
         """ Starts the nio Service. """
-        self._instance._get('services/{}/start'.format(self._name))
+        self.command('start')
 
     def stop(self):
         """ Stops the nio Service. """
-        self._instance._get('services/{}/stop'.format(self._name))
+        self.command('stop')
+
+    def command(self, command, block=None, **request_kwargs):
+        '''send a command to the service or to the block.
+        To the service:
+            service.command('command')
+        To a block in the service:
+            service.command(block, 'command')
+
+        kwargs are passed onto the request. Some of use are:
+            data: add data onto the request
+            timeout: set the request timeout
+        '''
+        get = self._instance._get
+        cmd_structure = '{}/{}/{}'.format
+        if block is None:
+            return get(cmd_structure('services', self._name, command),
+                       **request_kwargs)
+        else:
+            return get(cmd_structure('blocks', block._name, command),
```

Comment:
This isn't the correct syntax for a block command. It's services/{service_name}/{block_name}/{command_name}.